Documentation/tutorial/intro_template.md

```diff
@@ -0,0 +1,170 @@
+Introduction to *DocFX Template System*
+===============================
+
+*DocFX Template System* provides a flexible way of defining and using templates.
+
+As the following DocFX workflow shows,
+
+![DocFX workflow](../spec/images/docfx_workflow_highlevel.png)
+
+DocFX loads the set of files and transforms them into different data models using different type of document processors. Afterwards, *DocFX Template System* loads these data models, and transforms them into output files based on the document type of the data model.
+
+Each document belongs to a document type. For example, the document type for Markdown files is `conceptual`, and the document type for `toc.md` files is `Toc`.
+
+For a specific *Template*, each document type can have several *Renderer File*s. For a specific document, *DocFX Template System* picks the corresponding *Renderer File*s to render the data model into output files.
+
+*Renderer File*
+------------------
+*Renderer File*s are template files written in a specific templating language. Currently DocFX supports the following templating languages:
+1. [Mustache](http://mustache.github.io) templating language
+2. [Liquid](https://github.com/Shopify/liquid) templating language
+
+### *Renderer File* in Mustache syntax
+
+#### Naming Rule
+*Renderer File*s in [Mustache](http://mustache.github.io) syntax **MUST** end with `.tmpl` extension. [Mustache Partial](http://mustache.github.io/mustache.5.html#partials) files **MUST** end with `.tmpl.partial`.
+
+The naming rule for the *Renderer File* is:
+`<document_type>.<extension>[.primary].tmpl`.
```

Comment:
It is the naming rule for Renderer file, I'd like to take Renderer file as the core file, and Preprocessor as a supplement to Renderer file. 

---
In reply to: [63292790](https://github.com/dotnet/docfx/pull/306#discussion_r63292790) [](ancestors = 63292790)